OrbitClientData/TracepointData.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitClientData/TracepointData.h"
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadConstants.h"
+
+using orbit_client_protos::TracepointEventInfo;
+
+void TracepointData::EmplaceTracepointEvent(uint64_t time, uint64_t tracepoint_hash,
+                                            int32_t process_id, int32_t thread_id, int32_t cpu,
+                                            bool is_same_pid_as_target) {
+  absl::MutexLock lock(&mutex_);
+  num_total_tracepoints_++;
+
+  orbit_client_protos::TracepointEventInfo event;
+  event.set_time(time);
+  CHECK(HasTracepointKey(tracepoint_hash));
+  event.set_tracepoint_info_key(tracepoint_hash);
+  event.set_tid(thread_id);
+  event.set_pid(process_id);
+  event.set_cpu(cpu);
+
+  int32_t insertion_thread_id =
+      (is_same_pid_as_target) ? thread_id : orbit_base::kNotTargetProcessThreadId;
+
+  auto [event_map_iterator, unused_inserted] =
+      thread_id_to_time_to_tracepoint_.try_emplace(insertion_thread_id);
+  auto [unused_iterator, event_inserted] =
+      event_map_iterator->second.try_emplace(time, std::move(event));
+  if (!event_inserted) {
+    ERROR(
+        "Tracepoint event was not inserted as there was already an event on this time and "
+        "thread.");
+  }
+}
+
+[[nodiscard]] const std::map<uint64_t, orbit_client_protos::TracepointEventInfo>&
+TracepointData::GetTracepointsOfThread(int32_t thread_id) const {
+  absl::MutexLock lock(&mutex_);
+  static std::map<uint64_t, orbit_client_protos::TracepointEventInfo> empty;
+  const auto& it = thread_id_to_time_to_tracepoint_.find(thread_id);
```

Comment:
seems like there are multiple problems with this method. the returned map can be modified while the caller is reading it because read is not guarder after this method is called.

I would consider dropping this method and replacing the call with one of ForEach methods.

Reply:
Turns out, it is unused anyways. I'll drop it completely.